File: jiadaizhao_LeetCode/0401-0500/0405-Convert a Number to Hexadecimal/0405-Convert a Number to Hexadecimal.py
class Solution:
    def toHex(self, num: 'int') -> 'str':
        if num == 0:
            return '0'
        table = '0123456789abcdef'
        result = []
        for i in range(8):
            result.append(table[num&15])
            num //= 16
            if num == 0:
                break
        return ''.join(result[::-1]).lstrip('0')

File: jiadaizhao_LeetCode/0401-0500/0405-Convert a Number to Hexadecimal/0405-Convert a Number to Hexadecimal.cpp
#include <bits/stdc++.h>
#include <vector>
#include <iostream>
#include <string>
#include <map>
#include <set>
#include <unordered_map>
#include <algorithm>
using namespace std;

class Solution {
public:
    string toHex(int num) {
        if (num == 0) {
            return "0";
        }
        unsigned int n = num;
        string result;
        while (n) {
            int temp = n % 16;
            if (temp > 9) {
                result += 'a' + (temp - 10);
            }
            else {
                result += '0' + temp;
            }
            n /= 16;
        }
        
        reverse(result.begin(), result.end());
        return result;
    }
};
